### PR TITLE
Programming Exercise Domain Commands

### PR DESCRIPTION
**Pull request by thilo-behnke**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: the project builds without code style warnings.
- [x] I updated the documentation and models.
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~I added (end-to-end) test cases for the new functionality.~

### Motivation and Context
It would help the tas/instructors to insert tasks and testCases from domain commands from the markdown editor.

### Description
The list of test cases is taken from template participation, if there is no template participation due to an error, the test case list is empty.

To make the test case list work, I had to define a new domainCommand type "MultiOptionCommand", this class can be used to construct dropdowns in the editor.
In the markdown-editor html template a class type check is done to differentiate between domainTagCommands and multiOptionCommands.

This PR also fixes the issue with the header dropdown submitting the programming-exercise forms.

### Steps for Testing

1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Select course with a programming exercise
4. Open the exercise via edit, or create/link a new programming exercise -> commands should work as expected

### Screenshots
image

### CODE REVIEW

File: src/main/webapp/app/entities/lecture/lecture-attachments.component.ts [hunk 1]
```diff
@@ -131,19 +131,22 @@ export class LectureAttachmentsComponent implements OnInit {
 
     downloadAttachment(downloadUrl: string) {
```
Comment (thilo-behnke) [inline src/main/webapp/app/entities/lecture/lecture-attachments.component.ts]: I have not changed this file and can't remove this change by checking it out again from develop - must be our pre-commit hook.

File: src/main/java/de/tum/in/www1/artemis/domain/Repository.java [hunk 2]
```diff
@@ -1,20 +1,23 @@
 package de.tum.in.www1.artemis.domain;
```
Comment (thilo-behnke) [inline src/main/java/de/tum/in/www1/artemis/domain/Repository.java]: Not part of my PR, autoformatting.

File: src/main/java/de/tum/in/www1/artemis/web/rest/RepositoryResource.java [hunk 3]
```diff
@@ -1,16 +1,21 @@
 package de.tum.in.www1.artemis.web.rest;
```
Comment (thilo-behnke) [inline src/main/java/de/tum/in/www1/artemis/web/rest/RepositoryResource.java]: Not part of my PR, autoformatting.

File: src/main/java/de/tum/in/www1/artemis/domain/Repository.java (hunk 2 above)
Comment (thilo-behnke) [reply src/main/java/de/tum/in/www1/artemis/domain/Repository.java]: My bad, files from the file-browser were not formatted...

File: src/main/java/de/tum/in/www1/artemis/web/rest/RepositoryResource.java (hunk 3 above)
Comment (thilo-behnke) [reply src/main/java/de/tum/in/www1/artemis/web/rest/RepositoryResource.java]: My bad, java files from the file-browser enhancements were not formatted...

**Comment by krusche** [comment]
I merged another PR on programming exercises first and now we have conflicts here. Could you please resolve them? Thanks!

**Comment by thilo-behnke** [comment]
Develop was merged and I did some smaller cleanup, but this somehow broke the bamboo build. I'll find out what causes the issue.

**Comment by krusche** [comment]
Bamboo build works again so we can merge this, right?

**Comment by krusche** [comment]
@thilo-behnke After merging your other PR, we again have merge conflicts 😔
Could you please fix them again? Also in case everything is done from your side, please change remove the tag work-in-progress and add the tag ready for review

**Comment by thilo-behnke** [comment]
@krusche All is merged and up-to-date now :)

**Pull request merged**